backend/services/simulation_engine.py

```
@@ -362,6 +389,158 @@ def _format_conversation_for_ai(self, conversation_history: List[Dict[str, str]]
+                message=user_message,
+                persona_data=persona_data,
+                scene_data=scene_data,
+                conversation_history=conversation_history,
+                user_progress_id=user_progress_id,
+                scene_id=scene_id
+            )
+            
+            # Calculate quality score (simplified for now)
+            quality_score = 0.8  # LangChain responses are generally higher quality
+            
+            return response, quality_score
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)